### PR TITLE
feat!: remove native components from public API

### PR DESCRIPTION
**Pull request by maciekstosio**
## Description

This PR removes native components from our API. The motivation is that those shouldn't be available directly, only through our interface. 

## Checklist

- [x] Ensured that CI passes

### CODE REVIEW

**Review by kkafar — COMMENTED** [review]
One thing that concerns me is whether platform extensions, e.g. `.web.tsx` / `.native.tsx` are supported by all major bundlers and not only Metro. Do we have knowledge here? 

Besides this I think we're good. Just make sure that we work as regular on fresh installation. Thanks!

File: src/index.tsx [hunk 1]
```diff
@@ -39,28 +30,12 @@ export {
...
-
-export {
-  default as ScreenContentWrapper,
-  NativeScreenContentWrapper,
-} from './components/ScreenContentWrapper';
-
-/**
- * Modules
- */
-export { default as NativeScreensModule } from './fabric/NativeScreensModule';
```
[the remark this answers is not in the document]
Comment (maciekstosio) [reply src/index.tsx:63]: It is not used anywhere, so I removed that as well. Imo all functions from it should go through our proxy. But I can revert this change.

File: src/components/ScreenContentWrapper.web.tsx [hunk 2]
```diff
@@ -1,6 +1,3 @@
 import { View } from 'react-native';
 
-export const NativeScreenContentWrapper = View;
-
-const ScreenContentWrapper = View;
-export default ScreenContentWrapper;
+export default View;
```
[the remark this answers is not in the document]
Comment (maciekstosio) [reply src/components/ScreenContentWrapper.web.tsx]: Changed in: https://github.com/software-mansion/react-native-screens/pull/2412/commits/4bc520d44421b6d7f3932bbfaee7d309811b78bc

**Comment by maciekstosio** [comment]
> One thing that concerns me is whether platform extensions, e.g. `.web.tsx` / `.native.tsx` are supported by all major bundlers and not only Metro. Do we have knowledge here?
> 
> Besides this I think we're good. Just make sure that we work as regular on fresh installation. Thanks!

Is there any other alternative then Re.Pack? Based on https://github.com/callstack/repack/blob/f119ab3eb94eff9d2cc1aec8fcf9f835c3025abc/website/src/public/configs/webpack.config#L77 I assume it handles this case.

**Review by kkafar — APPROVED** [review]
Looks good

**Pull request merged**